src/ET_Client.php

```diff
@@ -379,12 +455,19 @@ function __doRequest($request, $location, $saction, $version, $one_way = 0)
 	{
 		$doc = new DOMDocument();
 		$doc->loadXML($request);
-		$objWSSE = new WSSESoap($doc);
-		$objWSSE->addUserToken("*", "*", FALSE);
-		$this->addOAuth($doc, $this->getInternalAuthToken($this->tenantKey));
-				
-		$content = $objWSSE->saveXML();
-		$content_length = strlen($content);
+
+        if($this->useOAuth2Authentication === true){
+            $this->addOAuth($doc, $this->getAuthToken($this->tenantKey));
```

Comment:
can you explain this change?

Reply:
The legacy authentication in PHP uses the internalAuthToken and the oAuth header. For the OAuth2, we need to use the authToken and set fueloauth in the headers.